server.go

```diff
@@ -524,15 +532,14 @@ func (c *serverConn) run(sctx context.Context) {
 			// TODO(stevvooe): Not wildly clear what we should do in this
 			// branch. Basically, it means that we are no longer receiving
 			// requests due to a terminal error.
-			recvErr = nil // connection is now "closing"
-			if err == io.EOF || err == io.ErrUnexpectedEOF {
+			opErr, isOpErr := err.(*net.OpError)
```

Comment:
What's the purpose of this type check? Shouldn't we just be able to use `errors.Is` without this?

Reply:
https://go.dev/play/p/fzZpVtJdorj